### PR TITLE
feat: added hyper_v_generation parameter to data_disks

### LINKED ISSUE #29: add the hyper_v_generation parameter to data disks
We have a virtual machine with an older data disk that needs its data disk hyper_v_generation parameter set to `v1`. This module in its current state will set the value to `null` and therefore force the data disk to be replaced by terraform.

### PR DESCRIPTION
**Pull request by stiross79**
Closes #29

### CODE REVIEW

**Comment by hknutsen** [comment]
Guess this depends on the resolution in #28.

File: main.tf [hunk 1]
```diff
@@ -163,8 +163,8 @@ resource "azurerm_managed_disk" "this" {
   create_option        = "Empty"
   disk_size_gb         = each.value.disk_size_gb
   storage_account_type = each.value.storage_account_type
-
-  tags = var.tags
+  hyper_v_generation   = each.value.hyper_v_generation
+  tags                 = var.tags
```
Comment (hknutsen) [inline main.tf]: Restore intentional whitespace:

```suggestion
  hyper_v_generation   = each.value.hyper_v_generation
  
  tags = var.tags
```

File: variables.tf [hunk 2]
```diff
@@ -144,6 +144,7 @@ variable "data_disks" {
     storage_account_type = optional(string, "Standard_LRS")
     caching              = optional(string, "ReadWrite")
     lun                  = optional(number)
+    hyper_v_generation   = optional(string, null)
```
Comment (hknutsen) [inline variables.tf]: `optional` sets the default value to `null` by default, so no need to set it explicitly:

```suggestion
    hyper_v_generation   = optional(string)
```

**Review by hknutsen — DISMISSED** [review]

File: variables.tf (hunk 2 above)
Comment (stiross79) [reply variables.tf]: You are correct. No need for default null values.

**Review by stiross79 — COMMENTED** [review]
Changes looks good.

**Review by helenakallekleiv — APPROVED** [review]
🚀

**Pull request merged**